capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -132,7 +132,9 @@ public static JsonObject createRequestBodyJson(HttpServletRequest request) throw
       return null;
     }
     String parameterValue = entry.getValue()[0];
-    Field[] fields = genericClass.getMyType().getDeclaredFields();
+    Class superClass = genericClass.getMyType().getSuperclass();
+    List<Field> fields = new ArrayList<>(Arrays.asList(genericClass.getMyType().getDeclaredFields()));
```

Comment:
Have you tried testing this with a class that doesn't have an explicit super-class?  Although, I think your implementation should work: I believe `superClass` will be `java.lang.Object` and `superClass.getDeclaredFields()` should return an empty list.

Reply:
I haven't tested it since in our case all classes at the very least extend BaseEntity (for ids). In general, I believe what you're saying is correct, since the `superClass` would be `java.lang.Object` which has no fields. The documentation says "If this Class object represents a class or interface with no declared fields, then this method returns an array of length 0." (https://docs.oracle.com/javase/10/docs/api/java/lang/Class.html#getDeclaredFields()) So in this case, appending the empty array wouldn't change anything, which is what we want. I could add an explicit check, but your reasoning was why I didn't.

Reply:
Sounds great, thanks for exploring the docs and clarifying the expected behavior here!